### PR TITLE
Support domain alias in fence module config

### PR DESCRIPTION
**Pull request by cywang1905**
## New feature
This feature supports adding the name of the dependent target service and a set of its regular matches, converted by a field `domainAlias`, to the sidecarscope. 

```proto
message Fence {
  // ...  
  // domain alias rule
  repeated DomainAlias domainAlias = 5;
}

message DomainAlias {
  string pattern = 1;
  repeated string templates = 2;
}
```

Supports both static and dynamic dependent target service.

## Usage example
Add `domainAlias` with single inverted commas in fence module config
```yaml
apiVersion: config.netease.com/v1alpha1
kind: SlimeBoot
metadata:
  name: lazyload
  namespace: mesh-operator
spec:
  module:
    - name: lazyload-test
      kind: lazyload
      enable: true
      general:
        wormholePort: # replace to your application service ports, and extend the list in case of multi ports
          - "9080"
        domainAliases: 
          - pattern: '(?P<service>[^\.]+)\.(?P<namespace>[^\.]+)\.svc\.cluster\.local$'
            templates:
              - "$namespace.$service.mailsaas"
  #...
```

serviceFence example
```yaml
apiVersion: microservice.slime.io/v1alpha1
kind: ServiceFence
metadata:
  name: ratings
  namespace: default
spec:
  enable: true
  host:
    details.default.svc.cluster.local: # static dependent service
      stable: {}
status:
  domains:
    default.details.mailsaas: # static dependent service converted result
      hosts:
      - default.details.mailsaas
    default.productpage.mailsaas: # dynamic dependent service converted result
      hosts:
      - default.productpage.mailsaas
    details.default.svc.cluster.local:
      hosts:
      - details.default.svc.cluster.local
    productpage.default.svc.cluster.local:
      hosts:
      - productpage.default.svc.cluster.local
  metricStatus:
    '{destination_service="productpage.default.svc.cluster.local"}': "1" # dynamic dependent service
```

sidecar example
```yaml
apiVersion: networking.istio.io/v1beta1
kind: Sidecar
metadata:
  name: ratings
  namespace: default
spec:
  egress:
  - hosts:
    - '*/default.details.mailsaas' # static dependent service converted result
    - '*/default.productpage.mailsaas' # dynamic dependent service converted result
    - '*/details.default.svc.cluster.local'
    - '*/productpage.default.svc.cluster.local'
    - istio-system/*
    - mesh-operator/*
  workloadSelector:
    labels:
      app: ratings
```

### CODE REVIEW

File: api/v1alpha1/fence_module.proto [hunk 1]
```diff
@@ -18,6 +18,10 @@ message Fence {
   repeated string namespace = 3;
   // custom outside dispatch traffic rules
   repeated Dispatch dispatches = 4;
+  // regexp expression, which is alias for target domain, format is 'pattern=template1,template2'
+  // default value is empty
+  // example: (?P<service>[^\.]+)\.(?P<namespace>[^\.]+)\.svc\.cluster\.local(?P<port>[^\.]+)?$=$namespace.$service.service.mailsaas$port
+  string domainAlias = 5;
```
Comment (YonkaFang) [inline api/v1alpha1/fence_module.proto:24]: use a structured data type

File: api/v1alpha1/fence_module.proto [hunk 2]
```diff
@@ -18,6 +18,10 @@ message Fence {
   repeated string namespace = 3;
   // custom outside dispatch traffic rules
   repeated Dispatch dispatches = 4;
+  // regexp expression, which is alias for target domain, format is 'pattern=template1,template2'
+  // default value is empty
+  // example: (?P<service>[^\.]+)\.(?P<namespace>[^\.]+)\.svc\.cluster\.local(?P<port>[^\.]+)?$=$namespace.$service.service.mailsaas$port
```
Comment (YonkaFang) [inline api/v1alpha1/fence_module.proto:23]: consider whether we allow port here

File: api/v1alpha1/fence_module.proto (hunk 1 above)
Comment (cywang1905) [reply api/v1alpha1/fence_module.proto:24]:
```proto
message Fence {
  // ...
  DomainAlias domainAlias = 5;
}

message DomainAlias {
  string pattern = 1;
  repeated string template = 2;
}

File: api/v1alpha1/fence_module.proto (hunk 2 above)
Comment (cywang1905) [reply api/v1alpha1/fence_module.proto:23]: no port  here

File: api/v1alpha1/fence_module.proto [hunk 3]
```diff
@@ -18,6 +18,8 @@ message Fence {
   repeated string namespace = 3;
   // custom outside dispatch traffic rules
   repeated Dispatch dispatches = 4;
+
+  DomainAlias domainAlias = 5;
```
Comment (YonkaFang) [inline api/v1alpha1/fence_module.proto:22]: repeated

Comment (cywang1905) [reply api/v1alpha1/fence_module.proto]: done

**Review by YonkaFang — APPROVED** [review]

**Pull request merged**